src/server/api/controllers/waypoints.controller.js

```diff
@@ -0,0 +1,82 @@
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+
+// Method for getting waypoints by id:
+const getWaypointsById = async id => {
+  try {
+    const waypoints = await knex("waypoints")
+      .select("id", "lon", "lat", "created_at", "updated_at")
+      .where({ id: id });
+    if (waypoints.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `Cannot find waypoints for ID ${id}!`,
+        409
+      );
+    }
+    return waypoints;
+  } catch (err) {
+    return err.message;
+  }
+};
+
+// Method for creating a waypoints
+
+const createWaypoints = async ({ body }) => {
+  const { lat, lon, port_id, route_id } = body;
+  console.log("this is body", body);
+
+  // 1. Get the route
+  const routes = await knex
+    .from("suggested_routes")
+    .select("*")
+    .where({ id: route_id });
+  if (routes.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `Cannot find routes for ID ${route_id}!`,
+      409
```

Comment:
Shouldn't the error be 404? Instead of 409!

Reply:
Yeah, 404 here and same further down.